Classes/Domain/Repository/HaikuRepository.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+
+class HaikuRepository
+{
+    private const TABLENAME = 'tx_examples_haiku';
+    private Connection $connection;
+
+    public function __construct(
+        ConnectionPool $connectionPool,
+    ) {
+        $this->connection = $connectionPool->getConnectionForTable(self::TABLENAME);
+    }
+
+    public function findAll(): array
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $result = $queryBuilder->select('*')->from(self::TABLENAME)->executeQuery();
+        return $result->fetchAllAssociative();
+    }
+
+    public function findByUid(int $uid): array
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        // $uid is an integer so we don't have to worry about SQL injections
+        $where = $queryBuilder->expr()->eq('uid', $uid);
+        $result = $queryBuilder->select('*')->from(self::TABLENAME)->where(
+            $where
+        )->executeQuery();
+        return $result->fetchAssociative() ?? [];
```

Comment:
fetchAssociative uses at the end [PDOStatement->fetch()](https://www.php.net/manual/de/pdostatement.fetch.php). With PDO::FETCH_ASSOC it returns an array or false (in case of an error). So the null coalescing operator does not make sense here. Either use `?:` to return an empty array in case of an error or throw an exception to make it clear that something is wrong. Or omit the fallback, then a TypeError is thrown for returning the wrong type.